model/src/main/kotlin/licenses/LicenseConfiguration.kt

```diff
@@ -19,61 +19,74 @@
 
 package org.ossreviewtoolkit.model.licenses
 
+import com.fasterxml.jackson.annotation.JsonAlias
 import com.fasterxml.jackson.annotation.JsonInclude
 
 /**
- * A configuration for licenses which allows to assign meta data to licenses. It allows assigning licenses to sets which
- * are useful for making license classifications. The available license sets are implicitly defined by the references
- * from the [licenses] and it is optional to define the meta data for any license set via an entry in [licenseSets].
+ * A classification for licenses which allows to assign meta data to licenses. It allows defining rather generic
+ * categories and assigning licenses to these. That way flexible classifications can be created based on
+ * customizable categories. The available license categories need to be declared explicitly; when creating an
+ * instance, it is checked that all the references from the [categorizations] point to existing [categories].
  */
 data class LicenseConfiguration(
     /**
-     * Defines meta data for the license sets.
+     * Defines meta data for the license categories.
      */
     @JsonInclude(JsonInclude.Include.NON_EMPTY)
-    val licenseSets: List<LicenseSet> = emptyList(),
+    @JsonAlias("license_sets")
+    val categories: List<LicenseCategory> = emptyList(),
 
     /**
      * Defines meta data for licenses.
      */
     @JsonInclude(JsonInclude.Include.NON_EMPTY)
-    val licenses: List<License> = emptyList()
+    @JsonAlias("licenses")
+    val categorizations: List<License> = emptyList()
 ) {
     init {
-        licenseSets.groupBy { it.id }.values.filter { it.size > 1 }.let { groups ->
+        categories.groupBy { it.name }.values.filter { it.size > 1 }.let { groups ->
             require(groups.isEmpty()) {
-                "Found multiple license set entries with the same Id: ${groups.joinToString { it.first().id }}."
+                "Found multiple license category entries with the same name: " +
+                        groups.joinToString { it.first().name }
             }
         }
-        licenses.groupBy { it.id }.values.filter { it.size > 1 }.let { groups ->
+
+        categorizations.groupBy { it.id }.values.filter { it.size > 1 }.let { groups ->
             require(groups.isEmpty()) {
                 "Found multiple license entries with the same Id: ${groups.joinToString { it.first().id.toString() }}."
             }
         }
+
+        val categoryNames = categories.map { it.name }.toSet()
+        categorizations.filterNot { it.categories.all(categoryNames::contains) }.let { lic ->
+            require(lic.isEmpty()) {
+                "Found licenses that reference non-existing categories: ${lic.joinToString { it.id.toString() }}"
+            }
+        }
```

Comment:
Sorry for not noticing this earlier, but the current code would break at the first categorization that refers to a non-existing category, but it's probably more convenient for the user to get to know all non-existing categories at once like

        val referredCategories = categorizations.mapTo(mutableSetOf()) { it.categories }
        val undefinedCategories = categories.mapTo(mutableSetOf()) { it.name } - referredCategories

        require(undefinedCategories.isEmpty()) {
            "Found licenses that reference non-existing categories: $undefinedCategories"
        }

Reply:
This check produces a different result. The original check generates a list of all license IDs that reference a non-existing category. Your variant now lists the unknown categories themselves. Maybe both lists are useful. I thought having the list of problematic licenses makes fixing the problem easier; from the list of unknown categories you would have to search the places where they are referenced.

Reply:
I was thinking that if there are unknown categories, the user either needs to add that category (so the category name is of importance to know here), or remove all references to that category from the file (which also requires to know the category name). From that perspective, I believe my output to be more useful to the user, would you agree?

Anyway, my main point really was to list all issues at once, and not stop at the first issue.

Reply:
The exception message now contains both a list of all licenses referencing unknown categories and the names of all unknown categories that are referenced. The test class checks whether really all names are contained.